Source/RunActivity/Viewer3D/Signals.cs

```diff
@@ -535,7 +533,20 @@ public SignalTypeData(Viewer viewer, SignalType mstsSignalType)
                             continue;
                         }
                         var mstsLight = viewer.SIGCFG.LightsTable[mstsSignalLight.Name];
-                        Lights.Add(new SignalLightPrimitive(viewer, new Vector3(-mstsSignalLight.X, mstsSignalLight.Y, mstsSignalLight.Z), mstsSignalLight.Radius, new Color(mstsLight.r, mstsLight.g, mstsLight.b, mstsLight.a), glowDay, glowNight, mstsLightTexture.u0, mstsLightTexture.v0, mstsLightTexture.u1, mstsLightTexture.v1));
+                        var material = defaultMaterial;
+                        if (!string.IsNullOrEmpty(mstsSignalLight.LightTextureName))
+                        {
+                            if (!viewer.SIGCFG.LightTextures.ContainsKey(mstsSignalType.LightTextureName))
```

Comment:
I believe this line is intended to be `if (!viewer.SIGCFG.LightTextures.ContainsKey(mstsSignalLight.LightTextureName))` as checking for `mstsSignalType.LightTextureName` allows line 545 to proceed even if the custom light texture is missing, which causes the signal to error out and not get rendered.

Replacing this line fixes the errors, causes the warnings to appear in the log, and results in signals appearing with no graphical issues.

```suggestion
                            if (!viewer.SIGCFG.LightTextures.ContainsKey(mstsSignalLight.LightTextureName))
```

Reply:
You're right, I copy-pasted the error handling and forgot to change it.